README.md

```diff
@@ -1,2 +1,58 @@
 # comp110-worksheet-C
 Base repository for COMP110 worksheet C
+```
+1: procedure HASDUPLICATE(list)
+2: 	let n be the length of list
+3: 		for i = 0, 1, . . . , n − 1 do
+4: 			for j = 0, 1, . . . , n − 1 do
+5: 				if i != j and list[i] = list[j] then
+6: 					return true
+7: 				end if
+8: 			end for
+9: 		end for
+10: 	return false
+11: end procedure
+
+```
+## a State what task this algorithm performs
+For each element in array i compare to each element in array j. If 2 elements are not in the same position in the list and they hold the same value then there is a duplicate, else there is not.
+***
+## b Explain why the worst case running time of the algorithm is quadratic, i.e.O(n^2).
+The search mamkes use of 2 liniar searches resulting in a quadratic search. This is due to the combination of nested loops resulting in the worst case being O(n^x) where x is the number of nested loops in this case 2. 
+***
+Now suppose that the for loop on line 4 is changed so that j ranges from 0 to i − 1
+***
+## c Explain why the algorithm is still correct.
+As the elements have to be in the same position to count as a duplicate there is no need to check any positions in array j after the position that the element holds in list i as it will always return as false.
```

Comment:
But we are explicitly comparing values in different positions in the list. The comparison won't always return false -- the key is that each comparison will be done twice (i.e. A==B and B==A)